app/models/institution.rb

```diff
@@ -161,7 +161,9 @@ class Institution < ActiveRecord::Base
   has_many :yellow_ribbon_programs, dependent: :destroy
   has_many :school_certifying_officials, -> { order 'priority, last_name' },
            primary_key: :facility_code, foreign_key: 'facility_code'
-  has_many :programs, -> { order(description: :asc) }, primary_key: :facility_code, foreign_key: 'facility_code'
+  has_many :institution_programs, -> { order(description: :asc) },
```

Comment:
You also need to account for `version` in this `has_many`.  i.e. the relationship is built on a compound key of `facility_code` and `version`

Reply:
Thanks!  This made me realize that I also made a mistake with the index on the migration PR (new PR coming soon...)